src/handlers/actions/EmitDomainReputation.ts

```diff
@@ -0,0 +1,43 @@
+import { BigNumber } from 'ethers';
+
+import { Id, getEvents } from '@colony/colony-js';
+import networkClient from '~/networkClient';
+import { ColonyActionType, ContractEvent } from '~/types';
+import { toNumber, writeActionFromEvent, getDomainDatabaseId } from '~/utils';
+
+export default async (event: ContractEvent): Promise<void> => {
+  const { contractAddress: colonyAddress } = event;
+  const { agent: initiatorAddress, user: userAddress, skillId, amount } = event.args;
+
+  const isReputationChangePositive = BigNumber.from(amount).gt(0);
+
+  const actionType = isReputationChangePositive ? ColonyActionType.EmitDomainReputationReward : ColonyActionType.EmitDomainReputationPenalty;
+
+  const colonyClient = await networkClient.getColonyClient(colonyAddress);
+  const domainAddedFilter = colonyClient.filters.DomainAdded(null, null);
+  const domainAddedEvents = await getEvents(colonyClient, domainAddedFilter);
+
+  const colonyDomains = await Promise.all(
+    domainAddedEvents.map(async (domain) => {
+      const domainId = domain.args.domainId.toString();
+      const { skillId } = await colonyClient.getDomain(domainId);
```

Comment:
Could this loop compare `skillId` as it goes and stop if it finds the right one? It doesn't seem like we need all the domains, just the one with the matching `skillId`.

Reply:
@jakubcolony  I couldn't find a simple way to do this because of the async call inside the map. So I added a new function to handle async calls inside a find.